### PR TITLE
Unity 2017 02 staging mbedtls

### PR DESCRIPTION
**Pull request by UnityAlex**

### CODE REVIEW

File: mcs/class/System/Mono.Mbedtls/MbedtlsContext.cs [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+	{
+		public readonly Mbedtls.mbedtls_ssl_context* Ptr;
+
+		public mbedtls_ssl_context_handle(Mbedtls.mbedtls_ssl_context* ptr)
+		{
+			Ptr = ptr;
+		}
+
+		public static implicit operator Mbedtls.mbedtls_ssl_context*(mbedtls_ssl_context_handle h)
+	    {
```
Comment (joncham) [inline mcs/class/System/Mono.Mbedtls/MbedtlsContext.cs]: Whitespace issues...

File: mono/utils/Makefile.am [hunk 2]
```diff
@@ -1,6 +1,6 @@
 noinst_LTLIBRARIES = libmonoutils.la
 
-AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/mono $(LIBGC_CPPFLAGS) $(GLIB_CFLAGS) $(SHARED_CFLAGS)
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/mono -I$(top_srcdir)/external/mbedtls/include $(LIBGC_CPPFLAGS) $(GLIB_CFLAGS) $(SHARED_CFLAGS)
```
Comment (joncham) [inline mono/utils/Makefile.am]: Do we want something like `MBEDTLS_CPPFLAGS` ?

File: mono/metadata/mbedtls.c [hunk 3]
```diff
@@ -0,0 +1,289 @@
+#ifdef HAVE_CONFIG_H
```
Comment (joncham) [inline mono/metadata/mbedtls.c]: I think we should name these two files `mono-mbedtls.h/c` to avoid potential conflicts.

File: msvc/mono.def [hunk 4]
```diff
@@ -998,3 +998,49 @@ mono_unity_oop_get_stack_frame_details
 ; Accessors for FATs in X64
 mono_unity_lock_dynamic_function_access_tables64
 mono_unity_unlock_dynamic_function_access_tables64
+
+
+mbedtls_entropy_init
```
Comment (joncham) [inline msvc/mono.def]: We should remove all these from the .def file now.

File: mono/utils/mono-systemcerts.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+typedef enum
+{
+	DATATYPE_STRING = 0,
+	DATATYPE_INTPTR = 1
+} CertDataFormat;
```
Comment (joncham) [inline mono/utils/mono-systemcerts.h]: We should prefix names with `Mono` to avoid conflicts.

File: mono/utils/mono-systemcerts.h [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+typedef struct
+{
+	void* certdata;
+	int certsize;
+} CertObj;
+
+int EnumSystemCertificates(void* certStore, void** iter, int *format, int* size, void** data);
```
Comment (joncham) [inline mono/utils/mono-systemcerts.h]: Let's not change these now, but when we move to IL2CPP the C API naming convention is `UnityPal*`

**Review by joshpeterson — APPROVED** [review]

**Comment by AndreasReich** [comment]
Closing PR. this is now obsolete with unitytls

**Pull request closed without merging**